policy/templates/subtemplates/auto/auto-test.gotmpl

```diff
@@ -19,65 +14,55 @@
+          [[ $VARIATION =~ test ]] && {
+              echo "::warning file=release.yml,line=100,col=5,endColumn=7::Using test variation"
+              echo "### :warning: You are using VARIATION with test in test-controller-{{ .test }}" >> $GITHUB_STEP_SUMMARY
+          }
 
-  {{ $test }}-tests:
+  {{ .test }}-tests:
     needs:
-    {{ if and (has "releng" $FEATURES) (ne $test "api-gate") (ne $test "ui-gate") }}
+      - test-controller-{{ .test }}
+    {{ if has "releng" .dot.Branchvals.Features }}
       - goreleaser
-    {{- end }}  
-      - test-controller-{{ $test }}
-    {{ if has "releng" $FEATURES }}
     runs-on: ubuntu-latest-m-2
     {{- else }}
     runs-on: ubuntu-latest
     {{- end }}
     env:
-      METADATA_REPORT_PATH: /tmp/metadata.toml
-      {{ if or (eq $test "ui") (eq $test "ui-gate") }}
-      XUNIT_REPORT_PATH: {{`${{ github.workspace }}`}}/tyk-analytics/tests/ui/playwright-report/results.xml
-      {{- end }}
-      {{ if or (eq $test "api") (eq $test "api-gate") }}
-      XUNIT_REPORT_PATH: {{`${{ github.workspace }}`}}/reports/pytest-report.xml
-      {{- end }}      
+      XUNIT_REPORT_PATH: {{`${{ github.workspace}}`}}/test-results.xml
     permissions:
       id-token: write # This is required for requesting the Github JWT
       contents: read # This is required for actions/checkout
     strategy:
       fail-fast: false
       matrix:
-        envfiles: {{`${{ fromJson(needs.test-controller-`}}{{ $test }}{{`.outputs.envfiles) }}`}}
-        {{- if ne $REPO "tyk-pump" }}
-        pump: {{`${{ fromJson(needs.test-controller-`}}{{ $test }}{{`.outputs.pump) }}`}}
+        envfiles: {{`${{ fromJson(needs.test-controller-`}}{{ .test }}{{`.outputs.envfiles) }}`}}
+        {{- if ne .dot.Name "tyk-pump" }}
+        pump: {{`${{ fromJson(needs.test-controller-`}}{{ .test }}{{`.outputs.pump) }}`}}
         {{ end }}
-        {{- if ne $REPO "tyk-sink" }}
-        sink: {{`${{ fromJson(needs.test-controller-`}}{{ $test }}{{`.outputs.sink) }}`}}
+        {{- if ne .dot.Name "tyk-sink" }}
+        sink: {{`${{ fromJson(needs.test-controller-`}}{{ .test }}{{`.outputs.sink) }}`}}
         {{ end }}
-        include:
-          - db: postgres15
-            markers: "and not sql"
-        {{- if and (ne $REPO "tyk-pump") (ne $REPO "tyk-sink") }}
+        {{- if and (ne .dot.Name "tyk-pump") (ne .dot.Name "tyk-sink") }}
         exclude:
           - pump: tykio/tyk-pump-docker-pub:v1.8
             sink: $ECR/tyk-sink:master
           - pump: $ECR/tyk-pump:master
             sink: tykio/tyk-mdcb-docker:v2.4
         {{- end }}
     steps:
+      {{- if eq .dot.Name "tyk-analytics" }}
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+          sparse-checkout: tests/{{ .test }}
+      {{- end }}
```

Comment:
@alephnull test/ui folder is always needed for repos in order to run UI tests, we need to change this conditional to be based on test name instead

Reply:
It is always available now.